src/Database/PostgreSQL/Consumers/Components.hs

```diff
@@ -186,14 +213,23 @@ spawnDispatcher
   -> MVar ()
   -> TVar (M.Map ThreadId idx)
   -> TVar Int
+  -> Maybe (TMVar Bool)
   -> m ThreadId
 spawnDispatcher ConsumerConfig{..} useSkipLocked cs cid semaphore
-  runningJobsInfo runningJobs =
+  runningJobsInfo runningJobs mIdleSignal =
   forkP "dispatcher" . forever $ do
     void $ takeMVar semaphore
-    loop 1
+    ifM (loop 1) (setIdle False) (setIdle True)
   where
-    loop :: Int -> m ()
+    setIdle :: forall m' . (MonadBaseControl IO m') => Bool -> m' ()
+    setIdle isIdle = case mIdleSignal of
+      Nothing -> return ()
+      Just idleSignal -> do
```

Comment:
You can remove the outer `do` here (i.e. just write `-> atomically $ do ...`).

Reply:
Yeah, right.